### PR TITLE
Batch full frame crop

### PR DESCRIPTION
**Pull request by shaundillon**
This moves us a step closer to 1 click galleries

## Improvements
- Detect large selections (Separate PR)

fullframe2

### CODE REVIEW

**Comment by paperboyo** [comment]
Woooo! 👏 

Does it:
- hide button if all images in a selection already have full frames? (but still shows it when at least one doesn’t)
- hide progress indicator (...) when all crops are done?

Here I’m not sure: because Clear selection has no keyboard shortcut (should be ESC?), and because Create full frames is so close to it – should Create full frames be a two-step button for safety? (just thinking about someone using it by mistake on a [large selection](https://github.com/guardian/grid/issues/1508).

Overall: bombastic and indeed a (big) step closer.

**Comment by shaundillon** [comment]
> Does it:

>hide button if all images in a selection already have full frames? (but still shows it when at least one doesn’t)
Nope, good shout. I'll see if I can add it to this PR.

>hide progress indicator (...) when all crops are done?
Yep!

**Comment by paperboyo** [comment]
> Detect large selections (Separate PR)

Limit would be entirely arbitrary, but given the [very real and dangerous pains](https://github.com/guardian/grid/issues/1508), maybe it should apply to **all** selections/operations?

Currently if thumbs haven’t shown themselves while making a selection, images won be selected! And then, operations done on that large selection won’t only be not applied to these missing images but also, often, fail partially. This leaves the systems in a state where one cannot know which part of the selection the operation was applied to (e.g. a downloaded zip will contain just a random selection of images from your initial selection; much more problematic for batch metadata operations).

**Comment by shaundillon** [comment]
@paperboyo I could add an arbitrary limit to this, but applying it to *all* selections is a much bigger job

**Comment by paperboyo** [comment]
Alrighty, no worries!

File: kahuna/public/js/main.js [hunk 1]
```diff
@@ -220,10 +218,24 @@ kahuna.run(['$rootScope', 'httpErrors',
             function($rootScope, httpErrors) {
 
     $rootScope.$on('events:error:unauthorised', () =>
-        $rootScope.$emit('track:event', 'Authentication', null, 'Error', null, { 'Error code': httpErrors.unauthorised.errorCode }));
+        $rootScope.$emit(
```
Comment (akash1810) [inline kahuna/public/js/main.js]: minor: this could be its own pr...

Comment (shaundillon) [reply kahuna/public/js/main.js]: Linting wouldn't pass without it :(.

It's a separate commit at least, so it should be easy to unpick if needed?

Comment (akash1810) [reply kahuna/public/js/main.js]: Sorry, I hadn't realised it was a whitespace change! Ignore me entirely!
